src/odb/include/odb/db.h

```diff
@@ -7524,6 +7528,38 @@ class dbPowerSwitch : public dbObject
   // User Code End dbPowerSwitch
 };
 
+class dbScanChain : public dbObject
+{
+ public:
+  void setLength(uint length);
+
+  uint getLength() const;
+};
+
+class dbScanInst : public dbObject
+{
+ public:
+  enum SCAN_INST_TYPE
+  {
+    OneBit,
+    ShiftRegister,
+    BlackBox
+  };
+  void setBits(uint bits);
+
+  uint getBits() const;
+};
+
+class dbScanPartition : public dbObject
+{
+ public:
+};
+
+class dbScanPin : public dbObject
+{
+ public:
+};
```

Comment:
There is no way to create these objects or get/set any fields.  dbScanInst & dbScanChain  are only slightly better.  I think you need to add some custom methods here.

Reply:
Ok so long as you feel you won't need to change the schema again